waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingServiceTest.java

```diff
@@ -355,4 +360,118 @@ public void panopticOperationsHandlerSetUgi() throws Exception {
     assertThat(result, containsInAnyOrder("ugi", "ugi2"));
   }
 
+  @Test(expected = NoPrimaryMetastoreException.class)
+  public void noPrimaryMappingThrowsException() {
+    when(metaStoreMappingFactory.newInstance(federatedMetastore)).thenReturn(metaStoreMappingFederated);
+    AbstractMetaStore unavailableMetastore = newFederatedInstance("name2", "thrift:host:port");
+    MetaStoreMapping unavailableMapping = mockNewMapping(false, "name2_");
+    when(metaStoreMappingFactory.newInstance(unavailableMetastore)).thenReturn(unavailableMapping);
+
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(federatedMetastore, unavailableMetastore), queryMapping);
+    service.primaryDatabaseMapping();
+  }
+
+  @Test(expected = NoPrimaryMetastoreException.class)
+  public void noPrimaryThrowsExceptionForUnmappedDatabase() {
+    when(metaStoreMappingFactory.newInstance(federatedMetastore)).thenReturn(metaStoreMappingFederated);
+    AbstractMetaStore unavailableMetastore = newFederatedInstance("name2", "thrift:host:port");
```

Comment:
Is `unavailable` the best name for this? The name to me suggests a metastore that is down, perhaps that is the intention here and I'm just missing something.

Reply:
Actually, after going through this test again, there is no need for that metastore. Only a list of non-primary metastores is needed, and a list with one element (federatedMetastore) is fine.